src/Anonymization/Anonymizer/Core/DateAnonymizer.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MakinaCorpus\DbToolsBundle\Anonymization\Anonymizer\Core;
+
+use Doctrine\DBAL\Platforms\AbstractMySQLPlatform;
+use Doctrine\DBAL\Platforms\SqlitePlatform;
+use MakinaCorpus\DbToolsBundle\Anonymization\Anonymizer\AbstractAnonymizer;
+use MakinaCorpus\DbToolsBundle\Attribute\AsAnonymizer;
+use MakinaCorpus\QueryBuilder\Query\Update;
+
+#[AsAnonymizer(
+    name: 'date',
+    pack: 'core',
+    description: <<<TXT
+    Anonymize a column by changing the date it contains.
+    You can either choose a 'min' and a 'max' date, case in which a random date will
+    be selected between these bounds, or alternatively set a 'delta' which must be
+    a valid date interval string (e.g. "1 week", "1 day 10 hours", ...).
+    You should set the 'format' (default: 'datetime') value, this anonymizator can
+    work with 'datetime' or 'date' formats.
+    TXT
+)]
+class DateAnonymizer extends AbstractAnonymizer
+{
+    /**
+     * @inheritdoc
+     */
+    public function anonymize(Update $update): void
+    {
+        $format = $this->options->get('format', 'datetime');
+        if (!\in_array($format, ['date', 'datetime'])) {
+            throw new \InvalidArgumentException(\sprintf("'format' value is invalid, expected 'date' or 'datetime', got '%s'.", $format));
+        }
+
+        $min = $max = null;
+        if ($min = $this->options->get('min')) {
+            $min = $this->parseDate($min, 'min');
+        }
+        if ($max = $this->options->get('max')) {
+            $max = $this->parseDate($max, 'max');
+        }
+        if (($min && !$max) || ($max && !$min)) {
+            throw new \InvalidArgumentException("You must specify both 'min' and 'max' boundaries.");
+        }
+        // @phpstan-ignore-next-line False positive detected.
+        if ($min && $max) {
+            if ($this->options->get('delta')) {
+                throw new \InvalidArgumentException("'delta' option cannot be specified if 'min' and 'max' are in use.");
+            }
```

Comment:
I'm not sure we did that kind of check for the other anonymizers. If we decide to be strict like that, I think we have to be consistent across all anonymizers.

Reply:
Agree and disagree at the same time, I agree we should be consistent, but right now since #97 is not implemented yet, we need some kind of validation. I would have preferred being stricter on validation everywhere until we overhaul the options API.

Reply:
I did not ask you to fix all other anonymizers :wink: I just wanted to mention the point.

Reply:
I planned to add this kind of validation for FloatAnonymizer and IntAnonymizer with #97.

It's ok to have it here for me waiting for this issue for me